packages/utils/src/lib/collect/implementation/mock/process-helper.mock.ts

```diff
@@ -0,0 +1,76 @@
+import { vi } from 'vitest';
+import { ProcessConfig } from '../execute-process';
+import { join } from 'path';
+
+const asyncProcessPath = join(__dirname, './execute-process.mock.mjs');
+
+/**
+ * Helps to get a async process runner config for testing.
+ *
+ * @param cfg
+ */
+export function getAsyncProcessRunnerConfig(
+  cfg: Partial<ProcessConfig> & {
+    throwError?: boolean;
+    interval?: number;
+    runs?: number;
+    outputPath?: string;
+  } = { throwError: false },
+) {
+  const outputPath = cfg?.outputPath || './out-async-runner.json';
+  const args = [asyncProcessPath];
+  cfg?.interval ? args.push(cfg?.interval + '') : args.push('10');
+  cfg?.runs ? args.push(cfg?.runs + '') : args.push('4');
+  cfg?.throwError ? args.push('1') : args.push('0');
+  args.push(outputPath);
```

Comment:
I really don't like how the ternary operator is being used here. It's supposed to return different expressions based on a condition (FP-style immutable logic). But you're using it here to conditionally execute statements, you're not assigning the result because the point is the side-effect of mutating the array. An `if`/`else` would fit better with such an imperative approach.

But this could also be easily refactored to use `? :` in a more semantic way:

```suggestion
  const args = [asyncProcessPath];
  args.push(cfg?.interval ? cfg.interval + '' : '10');
  args.push(cfg?.runs ? cfg.runs + '' : '4');
  args.push(cfg?.throwError ? '1' : '0');
  args.push(outputPath);
```

And can in fact be further simplified into a completely immutable version:

```suggestion
  const args = [
    asyncProcessPath,
    cfg?.interval ? cfg.interval + '' : '10',
    cfg?.runs ? cfg.runs + '' : '4',
    cfg?.throwError ? '1' : '0',
    outputPath,
  ];
```

Reply:
Thx! way cleaner now.